protocols/kad/src/handler_priv.rs

```diff
@@ -544,7 +547,7 @@ where
             future::Either::Right(p) => void::unreachable(p),
         };
 
-        if let ProtocolStatus::Unconfirmed = self.protocol_status {
+        if let ProtocolStatus::Unknown = self.protocol_status {
```

Comment:
Currently the `ProtocolStatus` is indifferent on whether the protocol support of the remote is outbound (client) or inbound (server) or both. I don't think there is any value in knowing whether the remote node is a client or not from the perspective of the local `ConnectionHandler`. The only thing that matters to `libp2p-kad` is whether the remote node is a server or not. Thus I don't think there is any value in switching from `ProtocolStatus::Unknown` to `ProtocolStatus::Cofirmed` in `on_fully_negotiated_inbound`. Does that make sense?